cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -306,7 +306,7 @@ common::stack_index_maps(
 //-----------------------------------------------------------------------------
 IndexMap::IndexMap(MPI_Comm comm, std::int32_t local_size)
     : _comm_owner_to_ghost(MPI_COMM_NULL), _comm_ghost_to_owner(MPI_COMM_NULL),
-      _comm_symmetric(MPI_COMM_NULL)
+      _comm_symmetric(MPI_COMM_NULL), _shared_indices(0)
```

Comment:
`AdjacencyList` doesn't have a default constructor.
Any alternative to avoid this weird initialization?

Reply:
Make it a `std::unique_ptr`.